### PR TITLE
WIP Examples

### PR DESCRIPTION
**Pull request by unclebob418**
I've created a schema and some data and started creating some queries. Not sure if I've put the .sql file in the right location but happy to move it. This is only for sql server at the moment.

I've been raising issues as I go and fixed a couple of minor things (like missing uuid type and bigdecimal not being "numeric")

From a usability point of view, so far it's been ok, but not spectacular. Error messages are helpful but frequently very very long due to the complexity of the types, so take a fair while to try to figure out the problem. Having to wrap a select in brackets 

One thing that's been a bit annoying is namespace issues, where different tables contain foreign keys for the same table. I've worked around this by stuffing things in objects, but I think it would work a lot better if the tables contained their column vals so you could write e.g. `users.userId` instead of awkward imports, but I'm not really sure how we could achieve this.

### CODE REVIEW

**Comment by unclebob418** [comment]
relates to #12 #13 #14 #15, need to add one for 16

**Comment by unclebob418** [comment]
@jdegoes or @adamgfraser can you please review?

File: zio-sql/sql/shop.sql [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    product_id uniqueidentifier not null,
+    quantity integer not null,
+    unit_price money not null
+)
+
+
+insert users
+    (usr_id, first_name, last_name, dob)
+values
+    ('60b01fc9-c902-4468-8d49-3c0f989def37', 'Ronald', 'Russell', '1983-01-05'),
```
Comment (jdegoes) [inline zio-sql/sql/shop.sql:51]: Just checking this is all fake data and not previously owned test data IP, right? 😉

Comment (unclebob418) [reply zio-sql/sql/shop.sql:51]: Wrote it all myself or randomly generated e.g. order details started life like this before I grouped them

```sql
insert into order_details
select top 5
    order_id,
    product_id,
    1 as quantity,
    price
from orders o cross apply(
select top 1 *
    from product_prices pp
    where pp.effective <= o.order_date
    order by effective desc, newid() ) a
order by newid()
```

**Comment by jdegoes** [comment]
Awesome work and identified some issues we have to work on! Will comment on the issues separately (one of them, I don't think we need to 'fix', at least at the type level).

Thank you!

**Pull request merged**